src/data.table.h

```diff
@@ -115,7 +115,7 @@ int checkOverAlloc(SEXP x);
 
 // forder.c
 int StrCmp(SEXP x, SEXP y);
-uint64_t dtwiddle(void *p, int i);
+uint64_t dtwiddle(const void *p, int i);
```

Comment:
this seems to introduce gcc warning
```
bmerge.c: In function 'bmerge_r':
bmerge.c:346:62: warning: pointer type mismatch in conditional expression
     uint64_t (*twiddle)(void *, int) = isInt64 ? &i64twiddle : &dtwiddle;
                                                              ^
bmerge.c:346:40: warning: ISO C forbids initialization between function pointer and 'void *' [-Wpedantic]
     uint64_t (*twiddle)(void *, int) = isInt64 ? &i64twiddle : &dtwiddle;
                                        ^~~~~~~
```
it was caugth by our by one of our strict CI jobs
```
(! grep "warning:" data.table.Rcheck/00install.out)
 bmerge.c:346:62: warning: pointer type mismatch in conditional expression
 bmerge.c:346:40: warning: ISO C forbids initialization between function pointer and ‘void *’ [-Wpedantic]
```